### PR TITLE
feat: Issue 120 update user backend

### LINKED ISSUE #120: Add updateUser GraqphQL mutation.
**Problem/Why**
---

The GraphQL endpoint does not have the ability to update user fields.

**Goal/What**
---

An API user should be able to update their username, email address, name, and password while logged into their account.

**Solution/How**
---

Create GraphQL endpoints for updating the user's information.

### To Do / Acceptance Criteria
-   [x] New graphql endpoint for updating user

### Review Checklist
-   An API user can update their:
  - [ ] Username
  - [ ] Password
  - [ ] Email address
  - [ ] Name
-   [ ] Appropriate unit tests

### PR DESCRIPTION
**Pull request by gmeben**
closes #120 

This prepares the backend for user updates via mutations on the GraphQL API

**Unit Testing**

1. On the cloned repo, start Lando `lando start`
2. Navigate to the backend directory `cd backend`
3. Run the unit tests `lando artisan test`

### CODE REVIEW

File: backend/app/Policies/UserPolicy.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        if (!$is_updating_own_user
+            && $user->can(Permission::UPDATE_USERS)
+        ) {
+            return true;
+        }
+
+        // TODO: Check if user can update user within own publication
+
+        return $is_updating_own_user;
+
```
Comment (wreality) [inline backend/app/Policies/UserPolicy.php:67]:
```suggestion
        //User is updating their own record.
        if ($user->id === $model->id) {
            return true;
        }
        //User has global permission to update users
        if ($user->can(Permission::UPDATE_USERS)) {
            return true;
        }

        // TODO: Check if user can update user within own publication
        // No explicit permission so return false.
        return false;
```
I feel like this is a little easier to digest.

File: backend/app/Providers/AuthServiceProvider.php [hunk 2]
```diff
@@ -15,6 +17,7 @@ class AuthServiceProvider extends ServiceProvider
      */
     protected $policies = [
         // 'App\Models\Model' => 'App\Policies\ModelPolicy',
+        User::class => UserPolicy::class
```
Comment (wreality) [inline backend/app/Providers/AuthServiceProvider.php:20]: I didn't realize this was a step for applying policies.  I like it. (It had been feeling a little too "magic" for my taste)

**Review by wreality — CHANGES_REQUESTED** [review]
@gmeben I only have one refactor to suggest.

File: backend/app/Providers/AuthServiceProvider.php (hunk 2 above)
Comment (gmeben) [reply backend/app/Providers/AuthServiceProvider.php:20]: Your feeling wasn't wrong! https://laravel.com/docs/8.x/authorization#policy-auto-discovery

**Review by wreality — APPROVED** [review]

**Pull request merged**